### DISCUSSION TITLE
Modifications to YOLOv8

### DISCUSSION THREAD

**Discussion by love-coding-lmy** [ultralytics/ultralytics#4326] [Q&A]
I added a custom module C3STR to the yaml file on YOLOv8, but now I am reporting an error AttributeError: Can't get attribute 'C3STR' on<module 'ultra tics. nn. modules. block' from 'D: soft anaconda envs AdelaiDet lib site packages ultra tics n modules block. py'>
May I ask how to solve it

**Comment by glenn-jocher** [comment]
@love-coding-lmy hello there! It seems like you've created a custom module `C3STR` and are trying to integrate it into YOLOv8. The error you're encountering indicates that the YOLOv8 framework is unable to recognize your custom module. This typically happens when the module is not properly defined or registered within the YOLOv8 codebase.

To resolve this issue, please ensure the following steps have been taken:

1. Your `C3STR` module is correctly defined in a Python file within the YOLOv8 directory structure.
2. The module is imported and registered in the necessary files so that YOLOv8 can recognize it when parsing the YAML configuration.
3. The YAML file is correctly referencing your custom module with the appropriate syntax.

Make sure that your custom module is accessible to the scripts that need to use it, and that there are no typos or path issues that could be causing the import error.

If you've checked these steps and the issue persists, you might want to review the documentation on how to extend YOLOv8 with custom modules for further guidance. 😊
